src/renderer/helpers/playlists.js

```diff
@@ -0,0 +1,41 @@
+export const SORT_BY_VALUES = {
+  DateAddedNewest: 'date_added_descending',
+  DateAddedOldest: 'date_added_ascending',
+  AuthorAscending: 'author_ascending',
+  AuthorDescending: 'author_descending',
+  VideoTitleAscending: 'video_title_ascending',
+  VideoTitleDescending: 'video_title_descending',
+  Custom: 'custom'
+}
+
+export function getSortedPlaylistItems(playlistItems, sortOrder, locale, reversed = false) {
+  if (sortOrder === SORT_BY_VALUES.Custom) {
+    return reversed ? playlistItems.toReversed() : playlistItems
+  }
+
+  return playlistItems.toSorted((a, b) => {
+    const first = !reversed ? a : b
+    const second = !reversed ? b : a
+    return compareTwoPlaylistItems(first, second, sortOrder, locale)
+  })
+}
+
+function compareTwoPlaylistItems(a, b, sortOrder, locale) {
+  switch (sortOrder) {
+    case SORT_BY_VALUES.DateAddedNewest:
+      return b.timeAdded - a.timeAdded
+    case SORT_BY_VALUES.DateAddedOldest:
+      return a.timeAdded - b.timeAdded
+    case SORT_BY_VALUES.VideoTitleAscending:
+      return a.title.localeCompare(b.title, locale)
+    case SORT_BY_VALUES.VideoTitleDescending:
+      return b.title.localeCompare(a.title, locale)
+    case SORT_BY_VALUES.AuthorAscending:
+      return a.author.localeCompare(b.author, locale)
+    case SORT_BY_VALUES.AuthorDescending:
+      return b.author.localeCompare(a.author, locale)
```

Comment:
As you already mentioned potential performance problems, using `Intl.Collator` instead of `localeCompare` should help with that a bit.

> When comparing large numbers of strings, such as in sorting large arrays, it is better to create an `Intl.Collator` object and use the function provided by its `compare()` method.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

```suggestion

  let collator

  if (
    sortOrder === SORT_BY_VALUES.VideoTitleAscending ||
    sortOrder === SORT_BY_VALUES.VideoTitleDescending ||
    sortOrder === SORT_BY_VALUES.AuthorAscending ||
    sortOrder === SORT_BY_VALUES.AuthorDescending
  ) {
    collator = new Intl.Collator([locale, 'en'])
  }

  return playlistItems.toSorted((a, b) => {
    const first = !reversed ? a : b
    const second = !reversed ? b : a
    return compareTwoPlaylistItems(first, second, sortOrder, collator)
  })
}

function compareTwoPlaylistItems(a, b, sortOrder, collator) {
  switch (sortOrder) {
    case SORT_BY_VALUES.DateAddedNewest:
      return b.timeAdded - a.timeAdded
    case SORT_BY_VALUES.DateAddedOldest:
      return a.timeAdded - b.timeAdded
    case SORT_BY_VALUES.VideoTitleAscending:
      return collator.compare(a.title, b.title)
    case SORT_BY_VALUES.VideoTitleDescending:
      return collator.compare(b.title, a.title)
    case SORT_BY_VALUES.AuthorAscending:
      return collator.compare(a.author, b.author)
    case SORT_BY_VALUES.AuthorDescending:
      return collator.compare(b.author, a.author)
```